lib/tasks/database_data.rake

```diff
@@ -9,6 +11,64 @@ namespace :database_data do
   # your local, qa, or stage environments respectively
   # These tasks should be run from the root directory of the application on your local machine
 
+  def download_from_s3
+    bucket = ENV.fetch('AWS_BUCKET', 'edu.psu.libraries.devteam.rmd-backup')
+    prefix = ENV.fetch('DB_PREFIX', 'db')
+    prompt = TTY::Prompt.new
+    client = Aws::S3::Client.new
+
+    download_prog = ProgressBar.create(total: nil, title: 'download', format: '%t |%B| %a')
+    objects = client.list_objects({ bucket: bucket, prefix: prefix }).contents.reverse
+    options = objects.map(&:key)
+    key = prompt.select('Choose your File', options)
+    filename = "#{Rails.root}/tmp/#{File.basename(key)}"
+    Thread.new { until download_prog.finished? do download_prog.increment; sleep(0.2); end }
+    client.get_object(response_target: filename, bucket: bucket, key: key)
+    download_prog.finish
+    puts "database downloaded to #{filename}"
+    filename
+  end
+
+  task download_from_s3: :environment do
+    desc 'Downloads a database dump from s3'
+    download_from_s3
+  end
+
+  task :load_from_backup, [:dump_file] => :environment do |_t, args|
+    desc 'Loads a database backup into the development environment'
+    filename = args[:dump_file]
+    db_config = Rails.configuration.database_configuration
+
+    filename ||= download_from_s3
+
+    # pg_restore expects these environment variables
+    ENV['PGHOST'] = db_config['development']['host']
+    ENV['PGUSER'] = db_config['development']['username']
+    ENV['PGPASSWORD'] = db_config['development']['password']
+    ENV['PG_DATABASE'] = db_config['development']['database']
+
+    # Remove any active connections to the database before purging
+    begin
+      ActiveRecord::Base.connection.execute <<-SQL.squish
+      SELECT pg_terminate_backend(pg_stat_activity.pid)
+      FROM pg_stat_activity
+      WHERE pg_stat_activity.datname = '#{db_config['development']['database']}';
+      SQL
+    rescue ActiveRecord::StatementInvalid
+      puts 'All connections killed.'
+    end
+
+    ActiveRecord::Tasks::DatabaseTasks.purge_current
+
+    psql_prog = ProgressBar.create(total: nil, title: 'restore', format: '%t |%B| %a')
+    Thread.new { until psql_prog.finished? do psql_prog.increment; sleep(0.2); end }
+    `pg_restore --no-owner --no-acl --dbname #{db_config['development']['database']} #{filename}`
+    psql_prog.finish
+
+    # Catch up on any migrations
+    ActiveRecord::Tasks::DatabaseTasks.migrate
+  end
+
   task get_prod_data_file: :environment do
```

Comment:
We can delete the tasks below now.  They don't work anymore; not since the server auth changes (or whatever happened).